statsmodels/tsa/statespace/tbats.py

```diff
@@ -0,0 +1,1430 @@
+
+    Parameters
+    ----------
+    periods:   iterable of float or None or array, optional
+        The period of the seasonal component. Default is None. Can be multiple
+    k:  int or None, optional, should be same length of periods
+    order:  iterable, optional
+        The (p,q) order of the model for the number of AR parameters, and MA parameters of error.
+    period_names:    iterable or None, optional
+        The names used for periods, should be the same length of periods
+
+    """
+
+    periods = PeriodWrapper('Seasonal Periods', 'periods', np.float, multiple=True)
+
+    def __init__(self, endog, periods=None, k=None, order=(0, 0), period_names=None,
+                 use_trend=True, damped_trend=True, box_cox=True, exog=None, mle_regression=True,
+                 **kwargs):
+
+        # Model options
+        self.seasonal = periods is not None
+        self.trend = use_trend
+        self.boxcox = box_cox
+        self.damping = damped_trend
+        self.mle_regression = mle_regression
+        self.autoregressive = (order != (0, 0))
+        self.periods = periods
+        self.k = k
+        self.order = order
+        k_states = 1
+
+        # Exogenous component
+        self.k_exog = 0
+        if exog is not None:
+            exog_is_using_pandas = _is_using_pandas(exog, None)
+            if not exog_is_using_pandas:
+                exog = np.asarray(exog)
+
+            # Make sure we have 2-dimensional array
+            if exog.ndim == 1:
+                if not exog_is_using_pandas:
+                    exog = exog[:, None]
+                else:
+                    exog = pd.DataFrame(exog)
+
+            self.k_exog = exog.shape[1]
+        self.regression = self.k_exog > 0
+
+        if periods is not None:
+            self.seasonal = True
+            self.k = k
+            k_states += int(2 * np.sum(self.k))
+            self.max_period = max(self.periods)
+            if period_names is not None:
+                self.period_names = period_names
+        else:
+            self.seasonal = False
+            self.max_period = 0
+            self.periods = (0,)
+            self.k = (0,)
+
+        self.trend = use_trend & True
+        if self.trend:
+            k_states += 1
+
+        self.autoregressive = np.sum(order) != 0
+        self.order = order
+
+        p, q = self.order
+
+        if self.autoregressive:
+            k_states += p + q
+
+        k_states = (
+            1 + self.trend +
+            np.sum(self.k) * 2 +
+            np.sum(self.order) +
+            (not self.mle_regression) * self.k_exog
+        )
+        k_posdef = 1
+
+        loglikelihood_burn = kwargs.get('loglikelihood_burn', int(self.max_period) + p + q)
+
+        # Initialize the model base
+        super(TBATSModel, self).__init__(endog=endog, exog=exog, k_states=k_states, k_posdef=k_posdef,
+                                         initialization='approximate_diffuse', **kwargs)
```

Comment:
It looks like you do a custom initialization below (which is great) - is this approximate diffuse specification ever used?